skfp/descriptors/constitutional.py

```diff
@@ -0,0 +1,252 @@
+from typing import Optional, Union
+
+from rdkit.Chem import AddHs, Mol
+from rdkit.Chem.Descriptors import MolWt
+from rdkit.Chem.rdMolDescriptors import CalcNumRings, CalcNumRotatableBonds
+
+
+def validate_molecule(func):
+    """
+    Decorator to validate that the molecule has at least one atom.
+
+    Raises
+    ------
+    ValueError
+        If the molecule has no atoms.
```

Comment:
This is an internal function, so it will suffice to just write `Validator for functions operating on single molecule. Ensures it is nonempty (at least 1 atom), raises ValueError otherwise`.